From b079ce539c4e28e1b30a7d6582651c131de91f17 Mon Sep 17 00:00:00 2001
From: Sam Hughes <sam@samuelhughes.com>
Date: Wed, 7 Jun 2023 13:45:13 -0700
Subject: [PATCH 1/2] Include stdint.h in errors.hpp and some other headers

context_switching.hpp
authentication_error.hpp

This addresses a specific alpine (musl) build failure.  We add the
include to errors.hpp to reduce the probability of such iwyu mishaps
in the future.
---
 src/arch/runtime/context_switching.hpp                      | 3 ++-
 src/clustering/administration/auth/authentication_error.hpp | 2 ++
 src/errors.hpp                                              | 1 +
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/arch/runtime/context_switching.hpp b/src/arch/runtime/context_switching.hpp
index 8a41c3db29f4..6f895fe0574a 100644
--- a/src/arch/runtime/context_switching.hpp
+++ b/src/arch/runtime/context_switching.hpp
@@ -35,9 +35,10 @@ void context_switch(fiber_context_ref_t *current_context_out, fiber_context_ref_
 typedef fiber_stack_t coro_stack_t;
 typedef fiber_context_ref_t coro_context_ref_t;
 
-#else
+#else  // _WIN32
 
 #include <pthread.h>
+#include <stdint.h>
 
 #include "errors.hpp"
 
diff --git a/src/clustering/administration/auth/authentication_error.hpp b/src/clustering/administration/auth/authentication_error.hpp
index 5c933caf2d37..3c1e287f5629 100644
--- a/src/clustering/administration/auth/authentication_error.hpp
+++ b/src/clustering/administration/auth/authentication_error.hpp
@@ -2,6 +2,8 @@
 #ifndef CLUSTERING_ADMINISTRATION_AUTH_AUTHENTICATION_ERROR_HPP
 #define CLUSTERING_ADMINISTRATION_AUTH_AUTHENTICATION_ERROR_HPP
 
+#include <stdint.h>
+
 #include <stdexcept>
 #include <string>
 
diff --git a/src/errors.hpp b/src/errors.hpp
index c45cef63c426..0ece10601f09 100644
--- a/src/errors.hpp
+++ b/src/errors.hpp
@@ -4,6 +4,7 @@
 
 #include <errno.h>
 #include <signal.h>
+#include <stdint.h>
 #include <stdlib.h>
 #include <string>
 
-- 
2.41.0

